src/com/sony/simdetect/SimDetectService.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019-2020 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rewritten in Kotlin by Pavel Dubrova <pashadubrova@gmail.com>
+ */
+
+package com.sony.simdetect
+
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.os.IBinder
+import android.os.Looper
+import android.os.Message
+import android.os.UEventObserver
+
+import com.android.internal.R
+import com.android.internal.telephony.uicc.UiccSlot
+
+class SimDetectService : Service() {
+    private var TAG = "SimDetectService"
+    // From drivers/misc/sim_detect.c
+    private var NOTHING_HAPPENED = "0"
+    private var SIM_REMOVED = "1"
+    private var SIM_INSERTED = "2"
+    // From src/java/com/android/internal/telephony/uicc/UiccSlot.java
+    private var EVENT_CARD_REMOVED = 13
+    private var EVENT_CARD_ADDED = 14
+
+    private val lock = Any()
```

Comment:
Looks like these got accidentally converted to `var` which is a mutable **variable**, rather than a `val` which is an immuatable **value**.